scalopus_tracing/src/lttng/lttng_source.cpp

```diff
@@ -131,10 +137,41 @@ std::vector<json> LttngSource::convertEvents()
         entry["ph"] = "i";
         entry["s"] = "t";
       }
+      else if (event.name() == "counter_event")
+      {
+        entry["ph"] = "C";
+        auto counter_series = LttngProvider::splitCounterSeriesName(trace_id_string);
+        entry["name"] = counter_series.first;
+        std::int64_t z = static_cast<std::int64_t>(event.eventData().at("value"));
+        // Update the current counters.
+        counter_values[counter_series.first][counter_series.second] = z;
+        entry["args"] = counter_values[counter_series.first];
+      }
       result.push_back(entry);
     }
   }
 
+  // Sort res by "ts" because sometimes the "E"nd events are out of order wrt the "B"egin, leading to unfinished scope
+  // events.
+  std::stable_sort(result.begin(), result.end(), [](const json& lhs, const json& rhs) {
+    return lhs.at("ts").get<double>() < rhs.at("ts").get<double>();
+  });
+
+  // Need a reverse iteration here, to populate all counters with all series seen in the entire interval.
+  CounterMap counte_all_series;
```

Comment:
`counte`?!

Reply:
Hah, typo and then copy pasted that without looking at it :)